lein2deps.clj

```diff
@@ -66,15 +66,18 @@ Options:
   --eval              : evaluate project.clj. Use at your own risk.")
     (let [project-clj (or (:project-clj opts)
                           "project.clj")
-          parsed (if (:eval opts)
-                   (load-file project-clj)
-                   (safe-parse project-clj))
-          {:keys [dependencies source-paths resource-paths compile-path java-source-paths]} parsed
+          project-edn (if (:eval opts)
+                        (load-file project-clj)
+                        (safe-parse project-clj))
+          project-edn (list* :compile-path "target/classes"
```

Comment:
project-edn is a map or seq of options but when it's a map something strange will happen here:

```
user=> (let [{:keys [foo]} (list* :foo 1 {:foo 2 :bar 2})] foo)
1
```

I suggest that the `defproject` macro should also return a map and that you would supply defaults using `:or` in destructuring or `merge`.

Reply:
Ah sorry, forgot to also try the non-eval case :sweat_smile: Seemed a bit fishy anyway, was surprised to find that `:keys` destructuring works on a flat kv sequence! Changed it now so that both code paths produce a map. Using `merge` for the defaults now since `:or` alone wouldn't pass the default `:compile-path` on to `add-prep-lib` via `project-edn`.